### PR TITLE
Add Jekyll support to website

### PR DESCRIPTION
**Pull request by ttpcodes**
This PR modifies the structure of the repository to add support for using features from `Jekyll`. For the most part, the actual contents of the website have been left alone. However, there have been changes to how some of the pages are laid out:
- `index.html`, `people.html`, `photos.html`, and `projects.html` have had their shared base template moved into a common file in `_layouts/main.html`, and any edits to the main layout should go there.
- `people.html` now automatically generates most of its content based off of the data in `_data/people.yml`, and any additions to the page should primarily go there.
- The `img/people/` directory has been refactored to separate different years in order to help organize the directory a bit more.

### CODE REVIEW

File: people.html [hunk 1]
```diff
@@ -48,647 +50,61 @@
...
+            {% for person in year.people %}
+              <div style="padding-right: 15px; padding-left: 15px; margin-bottom: 15px;">
+                <img src="img/people/{{ year.year }}/{{ person.kerberos }}.jpg" title="{{ person.kerberos }}" class="img-thumbnail" onerror="this.onerror = null; this.src = 'img/people/nopic-gourd.png'"/>
+                <h3>{{ person.name }} ({{ person.kerberos }}), '{{ year.year | slice: 2, 2 }}</h3>
+                <h4> Course {{ person.course }} </h4>
+                <p>{{ person.description | newline_to_br | markdownify }}</p>
+              </div>
+            {% endfor %}
+          </div>
+        {% endfor %}
```
Comment (cmnord) [inline people.html]: Blessed diff

File: people.html [hunk 2]
```diff
@@ -48,647 +50,61 @@
...
-              <h4> Course 6 </h4>
-              <p>Carolina and Jenny's roommate. Enjoys milk tea.</p>
-            </div>
-          </div>
-        </div>
+        {% for year in site.data.people.residents %}
+          <h3 class="section-heading">{{ year.year }}'s</h3>
+          <div class="resident-bios" style="display: grid; grid-template-columns: 1fr 1fr 1fr;">
+            {% for person in year.people %}
+              <div style="padding-right: 15px; padding-left: 15px; margin-bottom: 15px;">
```
Comment (cmnord) [inline people.html]: Can you separate styles out into their own css files?

File: _layouts/main.html [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        <div class="content">
+            {{ content }}
+        </div>
+        <!-- end main content -->
+    </div>
+</div>
+
+<footer>
+    <div class="container">
+        <p>2WebsiteE © 2017; Made with frozen grapes by residents of 2E <i id="grapes" class="em em-grapes"></i></p>
```
Comment (cmnord) [inline _layouts/main.html]: 2019? Or use liquid templating to current year

**Review by cmnord — APPROVED** [review]
LGTM overall; would like to see transition from HTML to markdown for main content in later PR

File: _layouts/main.html (hunk 3 above)
Comment (ttpcodes) [reply _layouts/main.html]: Probably best to manually update the copyright as the website is updated over the years.

File: people.html [hunk 4]
```diff
@@ -48,647 +50,61 @@
...
-              <h3>Lisa (lyang33), '17</h3>
-              <h4> Course 16 </h4>
-              <p>Fellow 2E Memer</p>
-            </div>
-          </div>
-        </div>
-
-        <h3 class="section-heading">2016's</h3>
+        <h2 class="heading">Friends</h2>
+        <hr style="margin: 0 0 10px 0"/>
```
Comment (cmnord) [inline people.html]: would also like to see this style pulled out

File: people.html [hunk 5]
```diff
@@ -1,127 +1,61 @@
...
-  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
-  <script src="js/main.js"></script>
-  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
-</body>
-
-</html>
+{% endfor %}
+
+<h2 class="heading">Alums</h2>
+<hr style="margin: 0 0 10px 0"/>
```
Comment (cmnord) [inline people.html]: (style note)

File: people.html [hunk 6]
```diff
@@ -1,127 +1,61 @@
...
+    <h3>{{ person.name }} ({{ person.kerberos }}), '{{ year.year | slice: 2, 2 }}</h3>
+    <h4> Course {{ person.course }} </h4>
+    <p>{{ person.description | newline_to_br | markdownify }}</p>
+  </div>
+  {% endfor %}
+</div>
+{% endfor %}
+
+<h2 class="heading">Friends</h2>
+<hr style="margin: 0 0 10px 0"/>
```
Comment (cmnord) [inline people.html]: (style note)

File: people.html [hunk 7]
```diff
@@ -1,127 +1,61 @@
...
+    <img src="img/people/friends/{{ person.kerberos }}.jpg" title="{{ person.kerberos }}" class="img-thumbnail" onerror="this.onerror = null; this.src = 'img/people/nopic-gourd.png'"/>
+    <h3>{{ person.name }} ({{ person.kerberos }}), '{{ person.year | slice: 2, 2 }}</h3>
+    <h4> Course {{ person.course }} </h4>
+    <p>{{ person.description | newline_to_br | markdownify }}</p>
+  </div>
+  {% endfor %}
+</div>
+
+<h2 class="heading">King Gourd</h2>
+<hr style="margin: 0 0 10px 0"/>
```
Comment (cmnord) [inline people.html]: (style note)

**Review by cmnord — APPROVED** [review]
lgtm :D

File: about.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
+---
+layout: page
```
Comment (cmnord) [inline about.md:2]: Can you delete this file?

File: 404.html [hunk 9]
```diff
@@ -0,0 +1,24 @@
+---
+layout: default
+---
+
+<style type="text/css" media="screen">
```
Comment (cmnord) [inline 404.html:5]: Nit: can this style be split out too? Or removed if not necessary

File: people.html [hunk 10]
```diff
@@ -1,711 +1,61 @@
...
-            </div>
-          </div>
-        </div>
+---
+layout: main
+---
+<h1>2WebsitE - Residents and Friends</h1>
+
+<h2 class="heading">Residents</h2>
+<hr style="margin: 0"/>
```
Comment (cmnord) [inline people.html:7]: Move style

**Review by cmnord — COMMENTED** [review]
U ask for more review; I give u more review

**Pull request merged**